### ISSUE TITLE
`(Rowan projectNamed: 'Rowan') gitRepositoryRoot: '$ROWAN_PROJECTS_HOME/Rowan'` is no longer sufficent to update gitRepositoryRoot of loaded project

### ISSUE THREAD

**Issue by dalehenrich** [GemTalk/Rowan#692]
@ericwinger hit a case today in recent 3.7.0 where the `projectsHome` field in RwGitRepositoryDefinitionV2 is brought over from the RwReadOnlyDiskRepositoryDefinitionV2 instance that extent0.rowan.dbf is built with is wired to the path that was appropriate for the build machine ... @ericwinger was attempting to load Rowan and the (new) RowanClientServices project choked when it referenced the build server path. Here's the relevant chunk of the stack:
```
16 RwPrjCloneTool >> _validateForGitRepository:ifDone: @10 line 31
    receiver aRwPrjCloneTool
    aRwGitRepositoryDefinition aRwGitRepositoryDefinitionV2
    doneBlock anExecBlock
    gitTool aRwGitTool
    response nil
    command nil
    cdResponse nil
    gitRepoPath /bcrux2/users/buildgss/37/build_73e37c375/externals_st/RowanClientServices
(skipped 2 evaluationTemps)
17 RwPrjCloneTool >> cloneRepository:            @2 line 4
    receiver aRwPrjCloneTool
    aRwGitRepositoryDefinition aRwGitRepositoryDefinitionV2
    gitTool nil
    response nil
    remoteUrl nil
    cloneOption nil
    checkout nil
(skipped 3 evaluationTemps)
18 RwGitRepositoryDefinitionV2 >> clone          @4 line 8
    receiver aRwGitRepositoryDefinitionV2
(skipped 2 evaluationTemps)
19 RwGitRepositoryDefinitionV2 >> resolve        @24 line 21
    receiver aRwGitRepositoryDefinitionV2
    url git@github.com:GemTalk/RowanClientServices.git
    gitTool nil
(skipped 1 evaluationTemps)
20 RwResolvedProjectV2 >> resolve                @3 line 5
    receiver aRwResolvedProjectV2
(skipped 1 evaluationTemps)
21 RwResolvedProjectV2 class >> loadSpecification: @3 line 8
    receiver RwResolvedProjectV2
    anRwLoadSpecificationV2 aRwLoadSpecificationV2
(skipped 1 evaluationTemps)
```
I'm thinking that the `projectsHome` field needs to be niled at the end of the build process (the step where we update the repositoryUrl to be `file:$GEMSTONE/upgrade/projects/RowanClientServices`.

The workaround is to reset projectsHome to $ROWAN_PROJECTS_HOME before doing build:
```
	(Rowan projectNamed: 'Rowan') _loadedProject handle projectsHome: '$ROWAN_PROJECTS_HOME'.
	(Rowan projectNamed: 'RowanClientServices') _loadedProject handle projectsHome: '$ROWAN_PROJECTS_HOME'.
```
While I'm at it add projectsHome setter/getter to RwProject ...

**Comment by dalehenrich** [comment]
While the above stack involves the class RwGitRepositoryDefinition ... upon further investigation it turns out that the load spec is the REAL source of the problem, since it is where the projectsHome is recorded ... and the load spec projectsHome is used to CREATE the RwGitRepositoryDefinition ... this happens because the project reference is resolved from the load spec on disk in RwResolvedProjectComponentVisitorV2 class >> readProjectSetForResolvedProject:withComponentNames:platformConditionalAttributes: ... At this point I'm thinking that the code in RwResolvedProjectComponentVisitorV2 class >> readProjectSetForResolvedProject:withComponentNames:platformConditionalAttributes: OUGHT to be seeing if the project exists in the image BEFORE running through the whole process of resolved the load spec ... 

On a side note, once a repo definition has been resolved (i.e., repositoryRoot defined), the projectsHome in the RwGitRepositoryDefinition is not really used (and/or shouldn't be used). So perhaps it should be niled out?

**Comment by dalehenrich** [comment]
fixed in b215aef
